fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java

```diff
@@ -320,13 +329,60 @@ public Table getTable(HiveTableName hiveTableName) {
 
     private Table loadTable(HiveTableName hiveTableName) throws TException, DdlException {
         org.apache.hadoop.hive.metastore.api.Table hiveTable = client.getTable(hiveTableName);
-        HiveTable table =  HiveMetaStoreTableUtils.convertToSRTable(hiveTable, resourceName);
+        Table table = null;
+        if (hiveTable.getSd().getInputFormat().contains("hudi")) {
+            Schema hudiTable = loadHudiTable(hiveTable);
+            table = HiveMetaStoreTableUtils.convertHudiConnTableToSRTable(hudiTable, hiveTable, resourceName);
+        } else {
+            table = HiveMetaStoreTableUtils.convertHiveConnTableToSRTable(hiveTable, resourceName);
+        }
         tableColumnStatsCache.invalidate(new HiveTableColumnsKey(hiveTableName.getDatabaseName(),
                 hiveTableName.getTableName(), null, null, table.getType()));
 
         return table;
     }
 
+    private Schema loadHudiTable(org.apache.hadoop.hive.metastore.api.Table hiveTable) throws DdlException {
+        Map<String, String> hudiProperties = Maps.newHashMap();
+        String hudiBasePath = hiveTable.getSd().getLocation();
+        if (!Strings.isNullOrEmpty(hudiBasePath)) {
+            hudiProperties.put("hudi.table.base.path", hudiBasePath);
+        }
+
+        Configuration conf = new Configuration();
+        HoodieTableMetaClient metaClient =
+                HoodieTableMetaClient.builder().setConf(conf).setBasePath(hudiBasePath).build();
+        HoodieTableConfig hudiTableConfig = metaClient.getTableConfig();
+
+        HoodieTableType hudiTableType = hudiTableConfig.getTableType();
+        if (hudiTableType == HoodieTableType.MERGE_ON_READ) {
+            throw new DdlException("MERGE_ON_READ type of hudi table is NOT supported.");
+        }
+        hudiProperties.put("hudi.table.type", hudiTableType.name());
+
+        Option<String[]> hudiTablePrimaryKey = hudiTableConfig.getRecordKeyFields();
+        if (hudiTablePrimaryKey.isPresent()) {
+            hudiProperties.put("hudi.table.primaryKey", hudiTableConfig.getRecordKeyFieldProp());
```

Comment:
I think you may use HudiTable.xxx to replace these strings.

Reply:
> I think you may use HudiTable.xxx to replace these strings.

done